sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs

```diff
@@ -278,16 +285,20 @@ private static void AddTelemetryFromActivityEvents(Activity activity, TelemetryI
                     exceptionType = tag.Value?.ToString();
                     continue;
                 }
-                if (tag.Key == SemanticConventions.AttributeExceptionMessage)
+                else if (tag.Key == SemanticConventions.AttributeExceptionMessage)
                 {
                     exceptionMessage = tag.Value?.ToString();
                     continue;
                 }
-                if (tag.Key == SemanticConventions.AttributeExceptionStacktrace)
+                else if (tag.Key == SemanticConventions.AttributeExceptionStacktrace)
                 {
                     exceptionStackTrace = tag.Value?.ToString();
```

Comment:
what happens when value is longer the max allowed limit? Would be common for stacktrace to hit that limit...
Would it result in whole telemetry being lost? Or we should trim to the limit, like we do for other general property bag?

It'd be nice to add this covered via tests, as we have seen Exceptions being lost due to lack of proper trimming in classic AI sdks..

Reply:
Same method, just a few lines lower...
Here the StackTrace gets truncated.

https://github.com/Azure/azure-sdk-for-net/blob/5c83c56d55343d3db9e245611d7715712531e6ed/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs#L309-L317

And yes, we have a unit test for the scenario where this is truncated.
https://github.com/Azure/azure-sdk-for-net/blob/5c83c56d55343d3db9e245611d7715712531e6ed/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryExceptionDataTests.cs#L76-L84